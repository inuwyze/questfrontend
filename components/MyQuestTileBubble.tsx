import React from "react"
import { View, Text, StyleSheet } from "react-native"
import { ITileBubbleProp } from "../types/QuestTypes"
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';


const MyQuestTileBubble:React.FC<ITileBubbleProp>=({title,description,creator,completed,navigation})=>{
    
    return(
        <View
        style={styles.container}>
        <View
        style={{flex:10}}>
            <Text
            style={styles.title}>
                {title}
            </Text>
            <Text>{creator}</Text>
            <Text>{description}</Text>
            {
                completed&&<View
                style={styles.completeStatus}>
                    <Text
                    
                    style={styles.completeStatusText}>
                        completed
                    </Text>

                </View>
            }
            </View>
            <View
            style={{
                flexDirection:'column',
                flex:1
            }}>
            <MaterialCommunityIcons name="chat" size={20} onPress={()=>navigation.navigate('ChatPage',{id:'rohan',text:'lets meet up',image:'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcS1V3C3qx89X6hSQZUNhi0QDhNWki6gD6B7Qg&usqp=CAU',number:"0987654324"})}/>
            </View>
        </View>
    )
}

const styles=StyleSheet.create({
    container:{    
        flexDirection:'row',
        justifyContent:'space-between'
        
    },
    title:{
        fontWeight:'bold',
        fontSize:20,
    },
    completeStatus:{
        padding:4,
        alignSelf:'flex-start',
        borderWidth:1,
        borderColor:'green',
        marginTop:10,  
    },
    completeStatusText:{
        color:'green'
    }
})

export default MyQuestTileBubble