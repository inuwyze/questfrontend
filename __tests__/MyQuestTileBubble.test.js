import React from 'react';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import MyQuestTileBubble from '../components/MyQuestTileBubble';


describe('Testing MyQuestTileBubble', () => {
  it('contains to View children', () => {
    const tree = renderer.create(<MyQuestTileBubble 
      title={'string'}
      description={'string'}
      creator={'string'}
      completed={'boolean'}
    />).toJSON();
    // console.log(tree)
    expect(tree.children.length).toBe(2);

  });
  
  it('renders all the passed props as texts', () => {
    const {getAllByText}=render(<MyQuestTileBubble 
      title={'QuestTile tile'}
      description={'some description'}
      creator={'henry'}
      completed={true}
    />)
      // console.log(r)
    expect(getAllByText('QuestTile tile').length).toBe(1)
    getAllByText('some description')
    getAllByText('henry')
  });
  
  it('Tile Bubble renders correctly', () => {
    const tree = renderer.create(<MyQuestTileBubble />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});