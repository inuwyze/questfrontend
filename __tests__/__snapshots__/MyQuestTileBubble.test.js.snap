// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MyQuestTileBubble Tile Bubble renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "flex": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    />
    <Text />
    <Text />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  />
</View>
`;
